### PR TITLE
implement couler codegen

### PR DESCRIPTION
**Pull request by Yancey0623**
part work of #1066 
TODO: add e2e test for couler mode

### CODE REVIEW

File: pkg/sql/executor_ir.go [hunk 1]
```diff
@@ -112,33 +114,57 @@ func submitWorkflow(wr *PipeWriter, sqlProgram string, db *DB, modelDir string,
...
+
 	// 2. compile Couler program into Argo YAML.
+	argoYaml, err := ioutil.TempFile("/tmp", "sqlflow-argo*.yaml")
+	cmd := exec.Command("couler", "run", "--mode", "argo", "--file", coulerFile.Name(), ">", argoYaml.Name())
+	if _, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("generate Argo workflow yaml error: %v", err)
+	}
+
 	// 3. submit Argo YAML and fetch the workflow ID.
+	cmd = exec.Command("kubectl", "create", "-f", argoYaml.Name())
```
Comment (typhoonzero) [inline pkg/sql/executor_ir.go:160]: can use k8s API here?

Comment (Yancey0623) [reply pkg/sql/executor_ir.go:160]: Sure, will import k8s package when implement `Fetch` gRPC interface, and I will rewrite here to use k8s API.

File: pkg/sql/codegen/couler/codegen.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"sqlflow.org/sqlflow/pkg/sql/ir"
+)
+
+// Run generates Couler program
+func Run(programIR ir.SQLProgram) (string, error) {
+	// TODO(yancey1989): fill session as env
+	r := &coulerFiller{}
+	for _, sqlIR := range programIR {
+		ss := &sqlStatment{}
+		switch sqlIR.(type) {
```
Comment (typhoonzero) [inline pkg/sql/codegen/couler/codegen.go:28]: Maybe we need to update this implement to #1272.

File: pkg/sql/codegen/couler/template.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const coulerTemplateText = `
+import couler.argo as couler
+{{ range $ss := .SQLStatements }}
+	{{if $ss.IsExtendedSQL }}
+couler.run_container(command='''repl -e "{{ $ss.OriginalSQL }}"''', image="{{ $ss.DockerImage }}")
+	{{else}}
+# TODO(yancey1989): 
+#	using "repl -parse" to output IR and
+#	feed to "sqlflow_submitter.{submitter}.train" to submite the job
+couler.run_container(command='''repl -e "{{ $ss.OriginalSQL }}"''', image="{{ $ss.DockerImage }}")
```
Comment (typhoonzero) [inline pkg/sql/codegen/couler/template.go:37]: Maybe add env as the session at here.

**Review by typhoonzero — DISMISSED** [review]

**Review by typhoonzero — APPROVED** [review]

File: pkg/sql/codegen/couler/template.go (hunk 3 above)
Comment (Yancey0623) [reply pkg/sql/codegen/couler/template.go:37]: You're right, will add it in the next PR.

**Pull request merged**